packages/edit-site/src/components/page-templates/dataviews-templates.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * External dependencies
+ */
+import removeAccents from 'remove-accents';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalHeading as Heading,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo, useCallback, useEffect } from '@wordpress/element';
+import { useEntityRecords } from '@wordpress/core-data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import Page from '../page';
+import Link from '../routes/link';
+import AddedBy from '../list/added-by';
+import { TEMPLATE_POST_TYPE } from '../../utils/constants';
+import { DataViews } from '../dataviews';
+import DEFAULT_VIEWS from './default-views';
+import { useResetTemplateAction } from './template-actions';
+import { unlock } from '../../lock-unlock';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+const EMPTY_ARRAY = [];
+const defaultConfigPerViewType = {
+	list: {},
+	grid: {},
```

Comment:
Can we use a special "block preview" field for the grid view as "media field"?

Reply:
We can try that, but I think we can skip it for first iteration and only support `list` view type. The PR is already quite big. Sounds good?

Reply:
I was about to suggest starting only with list view as well.

Reply:
I'm ok starting with just the list view.